File: hkccc08j1.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main() {
    vector<int> casper, natalie;
    int C, N, W, L; cin >> C;
    for (int i = 0; i < C; i++) {
        cin >> W >> L;
        casper.push_back(W * L);
    }
    cin >> N;
    for (int i = 0; i < N; i++) {
        cin >> W >> L;
        natalie.push_back(W * L);
    }
    sort(casper.begin(), casper.end());
    sort(natalie.begin(), natalie.end());
    if (casper[C - 1] > natalie[N - 1]) {
        cout << "Casper";
    } else if (natalie[N - 1] > casper[C - 1]) {
        cout << "Natalie";
    } else {
        cout << "Tie";
    }
    return 0;
}

File: hkccc08j1.py
c = int(input())
cBest = 0

for i in range(c):
    w, l = map(int, input().split())
    cBest = max(cBest, w * l)

n = int(input())
nBest = 0

for i in range(n):
    w, l = map(int, input().split())
    nBest = max(nBest, w * l)

if cBest > nBest:
    print("Casper")
elif nBest > cBest:
    print("Natalie")
else:
    print("Tie")